src/main/scala/mesosphere/marathon/core/storage/impl/zk/ZkStoreSerialization.scala

```diff
@@ -0,0 +1,51 @@
+package mesosphere.marathon.core.storage.impl.zk
+
+import java.time.OffsetDateTime
+
+import akka.http.scaladsl.marshalling.Marshaller
+import akka.http.scaladsl.unmarshalling.Unmarshaller
+import akka.util.ByteString
+import com.google.protobuf.MessageLite
+import mesosphere.marathon.core.storage.{ IdResolver, MarathonProto, MarathonState }
+import mesosphere.marathon.state.{ AppDefinition, PathId }
+
+case class ZkId(category: String, id: String, version: Option[OffsetDateTime]) {
+  private val bucket = id.hashCode % 16
+  def path: String = version.fold(s"/$category/$bucket/$id") { v =>
+    s"/$category/$bucket/$id/versions/$v"
+  }
+}
+
+case class ZkSerialized(bytes: ByteString)
+
+trait ZkStoreSerialization {
+  val maxVersions = 25
```

Comment:
Agreed, still trying to figure out how to do that effectively. This is only the default value, I'll make that more clear.